commit 2af36160b01ee1b6ceead47096ea7cc97c4e9501
Author: Samuel Just <sjust@redhat.com>
Date:   Tue Feb 27 17:52:27 2024 -0800

    mgr/DaemonServer: decouple adjust_pgs max from mon_osd_max_creating_pgs
    
    The next commit will remove mon_osd_max_creating_pgs as it doesn't
    behave correctly in the context of OSDMonitor.  Create a new mgr
    specific config to replace it here.
    
    Signed-off-by: Samuel Just <sjust@redhat.com>

diff --git a/src/mgr/DaemonServer.cc b/src/mgr/DaemonServer.cc
index 29a9acea533..7f9775c9352 100644
--- a/src/mgr/DaemonServer.cc
+++ b/src/mgr/DaemonServer.cc
@@ -2678,7 +2678,9 @@ void DaemonServer::send_report()
 void DaemonServer::adjust_pgs()
 {
   dout(20) << dendl;
-  unsigned max = std::max<int64_t>(1, g_conf()->mon_osd_max_creating_pgs);
+  uint64_t max = std::max<uint64_t>(
+    1,
+    g_conf().get_val<uint64_t>("mgr_max_pg_creating"));
   double max_misplaced = g_conf().get_val<double>("target_max_misplaced_ratio");
   bool aggro = g_conf().get_val<bool>("mgr_debug_aggressive_pg_num_changes");
 
